### PR TITLE
Fix the potential hazards of gmtime

### PR DESCRIPTION
**Pull request by Xiaohanjlll**
Fix: #1139 
Signed-off-by: Xiaohanjlll <hanx.xiao@intel.com>

### CODE REVIEW

**Review by liuw1 — COMMENTED** [review]
Why did you handle it for windows?

**Review by liuw1 — APPROVED** [review]

File: os_stub/cryptlib_mbedtls/sys_call/timer_wrapper_host.c [hunk 1]
```diff
@@ -15,13 +16,17 @@
 struct tm *mbedtls_platform_gmtime_r(const mbedtls_time_t *tt,
                                      struct tm *tm_buf)
 {
-    struct tm *lt;
-
-    lt = gmtime(tt);
+#ifdef _MSC_VER
+    if (gmtime_s(tm_buf, tt) != 0) {
```
Comment (steven-bellock) [inline os_stub/cryptlib_mbedtls/sys_call/timer_wrapper_host.c:20]: Should be `gmtime_s(tm_buf, tt) == NULL`, like below.

Comment (Xiaohanjlll) [reply os_stub/cryptlib_mbedtls/sys_call/timer_wrapper_host.c:20]: > Should be `gmtime_s(tm_buf, tt) == NULL`, like below.
errno_t gmtime_s(
   struct tm* tmDest,
   const __time_t* sourceTime
);
typedef int                           errno_t;
C4047: "errno_t" and "void *" have different indirection levels

basis:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/gmtime-s-gmtime32-s-gmtime64-s?view=msvc-170

Comment (steven-bellock) [reply os_stub/cryptlib_mbedtls/sys_call/timer_wrapper_host.c:20]: C11 defines it as `struct tm *gmtime_s(const time_t * restrict timer, struct tm * restrict result);` so I guess this compiler isn't following the standard? I guess in the short-term we'll go with the compiler.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**